src/main/java/com/synopsys/integration/alert/web/security/authentication/ldap/LdapManager.java

```diff
@@ -82,41 +81,45 @@ public boolean isLdapEnabled() {
         return false;
     }
 
-    public ConfigurationModel getCurrentConfiguration() throws AlertDatabaseConstraintException, AlertConfigurationException {
-        return configurationAccessor.getConfigurationsByDescriptorKey(authenticationDescriptorKey)
-                   .stream()
-                   .findFirst()
-                   .orElseThrow(() -> new AlertConfigurationException("Settings configuration missing"));
+    public FieldAccessor getCurrentConfiguration() throws AlertDatabaseConstraintException, AlertConfigurationException {
+        ConfigurationModel configModel = configurationAccessor.getConfigurationsByDescriptorKey(authenticationDescriptorKey)
+                                             .stream()
+                                             .findFirst()
+                                             .orElseThrow(() -> new AlertConfigurationException("Settings configuration missing"));
+        return new FieldAccessor(configModel.getCopyOfKeyToFieldMap());
     }
 
-    public LdapAuthenticationProvider getAuthenticationProvider() throws AlertConfigurationException {
-        updateContext();
-        return authenticationProvider;
+    public Optional<LdapAuthenticationProvider> getAuthenticationProvider() throws AlertConfigurationException {
+        try {
+            FieldAccessor fieldAccessor = getCurrentConfiguration();
+            return createAuthProvider(fieldAccessor);
+        } catch (IllegalArgumentException | AlertDatabaseConstraintException ex) {
```

Comment:
It doesn't look like there will ever be an `IllegalArgumentException` because we catch it in the method called.